pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -281,16 +285,34 @@ public long getLatestIngestionTimestamp() {
           invertedIndexColumns.contains(column) ? indexProvider.newInvertedIndex(context.forInvertedIndex()) : null;
 
       // Text index
-      RealtimeLuceneTextIndex textIndex;
+      MutableTextIndex textIndex;
+      boolean shouldBuildLuceneIndex = true;
       if (textIndexColumns.contains(column)) {
-        // TODO - this logic is in the wrong place and belongs in a Lucene-specific submodule,
-        //  it is beyond the scope of realtime index pluggability to do this refactoring, so realtime
-        //  text indexes remain statically defined. Revisit this after this refactoring has been done.
-        textIndex = new RealtimeLuceneTextIndex(column, new File(config.getConsumerDir()), _segmentName);
-        if (_realtimeLuceneReaders == null) {
-          _realtimeLuceneReaders = new RealtimeLuceneIndexRefreshState.RealtimeLuceneReaders(_segmentName);
+        if (_fieldConfigList != null) {
+          for (FieldConfig fieldConfig : _fieldConfigList) {
+            if (fieldConfig.getName().equalsIgnoreCase(column)) {
+              Map<String, String> properties = fieldConfig.getProperties();
+
+              if (TextIndexUtils.isFstTypeNative(properties)) {
+                shouldBuildLuceneIndex = false;
+              }
+            }
+          }
+        }
+
+        if (shouldBuildLuceneIndex) {
+          // TODO - this logic is in the wrong place and belongs in a Lucene-specific submodule,
+          //  it is beyond the scope of realtime index pluggability to do this refactoring, so realtime
+          //  text indexes remain statically defined. Revisit this after this refactoring has been done.
+          textIndex = new RealtimeLuceneTextIndex(column, new File(config.getConsumerDir()), _segmentName);
+          if (_realtimeLuceneReaders == null) {
+            _realtimeLuceneReaders = new RealtimeLuceneIndexRefreshState.RealtimeLuceneReaders(_segmentName);
+          }
+          _realtimeLuceneReaders.addReader((RealtimeLuceneTextIndex) textIndex);
+        } else {
+          //TODO : Add native index
```

Comment:
Log a warning, and fall back to the lucene one. We don't want to remove the index if it is mis-configured